packages/messenger/src/Messenger.ts

```diff
@@ -257,6 +287,11 @@ export class Messenger<
       );
     }
     this.#registerActionHandler(actionType, handler);
+    if (this.#parent) {
+      // @ts-expect-error The parent type isn't constructed in a way that proves it supports this
```

Comment:
I feel like there ought to be a way to align the types so this isn't necessary, but I don't know how. I tried changing the type of `#parent` from `DelegatedMessenger` to `Parent` but then the `if (messenger === this.#parent) {` statement you added at the beginning of `revoke` failed because TypeScript thought there was no overlap between the two values. So it seems that we need `@ts-expect-error` somewhere...

I do agree this seems type-safe though given the check we have in the constructor.